### PR TITLE
Removing generic hasher from block hash computation, using pedersen -…

### PR DESCRIPTION
**Pull request by elielnfinic**
This PR is related to issue  #1572

Here are the two actions that were performed : 
- Updated the hash() function from generic to support Pedersen
- Cleaned all the issues that resulted from that change.

# Pull Request type

Refactoring (no functional changes, no API changes)

## What is the current behavior?

Using the `hash` function located in `crates/primitives/block/src/headers.rs`, one can specify any hasher function.

Resolves: #NA

## What is the new behavior?

In this refactor, we can only use `pedersen` hasher function.

## Does this introduce a breaking change?

Yes but everything is fixed.

## Other information

NA

### CODE REVIEW

**Review by elielnfinic — COMMENTED** [review]
It should be ready for review.

File: crates/primitives/block/src/header.rs [hunk 1]
```diff
@@ -59,8 +60,8 @@ impl Header {
         }
     }
 
-    /// Compute the hash of the header.
-    pub fn hash<H: HasherT>(&self) -> Felt252Wrapper {
+    // Computing the hash using the Pedersen hasher.
```
[the remark this answers is not in the document]
Comment (tdelabro) [reply crates/primitives/block/src/header.rs:63]: Like

```
Compute the hash using the Pedersen hasher according to [the Starknet protocol specification](https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/header/#block_hash).
```

File: crates/pallets/starknet/src/lib.rs [hunk 2]
```diff
@@ -1007,7 +1007,7 @@ impl<T: Config> Pallet<T> {
         // Safe because it could only failed if `transaction_count` does not match `transactions.len()`
         .unwrap();
         // Save the block number <> hash mapping.
-        let blockhash = block.header().hash::<T::SystemHash>();
+        let blockhash = block.header().hash();
```
[the remark this answers is not in the document]
Comment (tdelabro) [reply crates/pallets/starknet/src/lib.rs:1010]: Yup, remove it wherever it is not used anymore

[the remark this answers is not in the document]
Comment (elielnfinic) [reply crates/pallets/starknet/src/lib.rs:1010]: Alright @tdelabro

**Review by tdelabro — APPROVED** [review]

**Pull request merged**